src/Compilers/Core/Portable/Syntax/InternalSyntax/SyntaxDiagnosticInfoList.cs

```diff
@@ -110,27 +103,26 @@ public bool MoveNext()
 
             private void PushNodeOrToken(GreenNode node)
             {
-                var token = node as SyntaxToken;
-                if (token != null)
+                if (node.IsToken)
```

Comment:
actually i'm going to eschew hte assert.  I have a objection to code that does ```Assert(x != null);  x.something```

```x.``` is already effectively asserting that it's not null.  So the assert doesn't do anything but move the crash one line higher.